File: Herencia/ejercicio 5/Java/Main.java
public class Main {
    public static void main(String[] args) {
        Gerente[] g = {
            new Gerente("Luis", "Gomez", 2000, 5, "Ventas", 1200),
            new Gerente("Ana", "Lara", 2200, 3, "RRHH", 800)
        };
        Desarrollador[] d = {
            new Desarrollador("Carlos", "Lopez", 1800, 2, "Java", 12),
            new Desarrollador("Laura", "Perez", 1700, 4, "Python", 8)
        };
        for (Gerente i : g) {
            System.out.println("Gerente: " + i.getNombre() + "Salario: " + i.calcularSalario());
            if (i.getBonoGerencial() > 1000) {
                System.out.println("Bono gerencial mayor a 1000: " + i.getBonoGerencial());
            }
        }
        for (Desarrollador j : d) {
            System.out.println("Desarrollador: " + j.getNombre() + "Salario: " + j.calcularSalario());
            if (j.getHorasExtras() > 10) {
                System.out.println("Trabaja más de 10 horas extras: " + j.getHorasExtras());
            }
        }
    }
}

File: Herencia/ejercicio 5/Python/Main.py
from Gerente import Gerente
from Desarrollador import Desarrollador
g = [Gerente("Luis", "Gomez", 2000, 5, "Ventas", 1200), Gerente("Ana", "Lara", 2200, 3, "RRHH", 800)]
d = [Desarrollador("Carlos", "Lopez", 1800, 2, "Java", 12), Desarrollador("Laura", "Perez", 1700, 4, "Python", 8)]
for i in g:
    print(f"Gerente: {i.nombre} - Salario: {i.calcular()}")
    if i.bono_gerencial > 1000:
        print(f"Bono gerencial mayor a 1000: {i.bono_gerencial}")
for j in d:
    print(f"Desarrollador: {j.nombre} - Salario: {j.calcular()}")
    if j.horas_extras > 10:
        print(f"Trabaja más de 10 horas extras: {j.horas_extras}")
